src/main/java/gift/domain/option/OptionService.java

```diff
@@ -82,4 +85,30 @@ public void updateOption(Long productId, Long optionId, OptionRequestDTO optionR
 
         optionRepository.save(option);
     }
+
+    // 상품 옵션 수량 차감
+    @Transactional
+    public void decreaseOptionQuantity(Long productId, Long optionId, Long quantity) {
+        Product product = productRepository.findById(productId)
+            .orElseThrow(() -> new ProductNotFoundException(productId));
+        Option option = optionRepository.findById(optionId)
+            .orElseThrow(() -> new OptionNotFoundException(optionId));
+
+        if (option.getQuantity() < quantity || quantity < 0) {
```

Comment:
이 부분도 다음과 같이 비즈니스 로직을 객체 안으로 넣어버리면 더욱 좋을 것 같아요ㅎㅎ
```suggestion
        if (option.canDecrease(quantity)) {
```

Reply:
추가적으로 삭제 부분이 다음과 같이 변경되어야 하지 않을까 싶어요!
테스트의 중요성, 테스트 중에서도 경계값 테스트의 중요성이 보이는 부분이네요!
```java
quantity <= 0
```

Reply:
앗 그러네요 수정하겠습니다! 
말씀대로 경계값 테스트를 수행하면 이러한 부분을 파악할 수 있어서 좋은 것 같습니다!